### DISCUSSION TITLE
euphony library에서 raw data를 extract

### DISCUSSION THREAD

**Discussion by sunyeongan** [euphony-io/euphony#214] [Q&A]
안녕하세요! 

https://github.com/orgs/euphony-io/projects/3/views/1
위의  대시보드를 읽고 질문이 생겨서 글 남깁니다! 

"euphony library가 음파를 raw data로 extract해주는 기능이 먼저 구현되어야 합니다." 

1. 이 부분에서 raw data는 16비트의 wave 데이터 인가요? 
2. 만약 16비트의 wave 데이터가 맞다면 이걸 pcm으로 변환하는 작업을 해야하나요?? 
3. 위의 코드를 어디 경로에 구현해야 하나요..?!

**Comment by designe** [comment]
안녕하세요! @sunyeongan 님!

1. 이 부분에서 raw data는 16비트의 wave 데이터 인가요?
https://github.com/euphony-io/euphony/blob/9c428075adfcb6f2c70eec11f995396a1dc85e63/euphony/src/main/java/co/euphony/rx/AudioRecorder.java#L27
위 코드에 명시된 것처럼 android의 AudioRecord API를 통해 16bit PCM data를 ByteBuffer 형식으로 받습니다.

2. 만약 16비트의 wave 데이터가 맞다면 이걸 pcm으로 변환하는 작업을 해야하나요??
입력값 : 16bit PCM data의 ByteBuffer 형식
출력값 : pcm 확장자 파일 (순수 raw data without header)
입력값이 이미 있다고 가정하고 클래스를 짜시면 좋을 것 같습니다. 최종적으로는 파일로 추출하면 됩니다.

3. 위의 코드를 어디 경로에 구현해야 하나요..?!
위의 역할을 하는 class를 co.jbear.euphony_hub_collector.util에 만들면 될 것 같아요. 
class 이름은 자유입니다. (가령, PCMFileCreator?) 
테스트 코드는 instrumental test까지 필요 없을 것 같고 일반 junit test로 하시면 편하실 것 같습니다 :)
